includes/core/classes/traits/class-block-variation.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Common class that handles everything a block-variation needs in GatherPress.
+ *
+ * @package GatherPress\Core
+ * @since 1.0.0
+ */
+
+namespace GatherPress\Core\Traits;
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit; // @codeCoverageIgnore
+
+use Error;
+
+/**
+ * Common class that handles registering and enqueuing of assets.
+ *
+ * @since 1.0.0
+ */
+trait Block_Variation {
```

Comment:
I think we can simplify this a bit and not include a new trait. I'd add code like this to `class-assets.php`, get the variations directory from the file system and loop through. There's a lot of helpers you can use in `class-assets.php` for getting dependencies and such from the asset. Let me know what you think.

Reply:
This sounds good.
Now that you mention the class-assets.php, I’m wondering that I haven’t thought about this. But yes, seems totally plausible for all this asset-related stuff.
Thank you.